src/routes/profile.js

```diff
@@ -0,0 +1,125 @@
+import express from 'express';
+import {getUserProfile, udpateProfile}  from '../controllers/profile';
+const router = express.Router();
+
+/**
+ * @swagger
+ * /user/profile/{userId}:
+ *    put:
+ *      summary: Updateing user profile
+ *      parameters:
+ *           - name: userId
+ *             in: path
+ *             required: true
+ *      tags: [Profile]
+ *      requestBody:
+ *        required: true
+ *        content:
+ *         application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/profile'
+ *      security:
+ *        - bearerAuth: []
+ *      responses:
+ *        "200":
+ *          description: Profile updated
+ *        "400": 
+ *          description: Bad request
+ *        "404":
+ *          description: Not found
+ *        "401":
+ *          description: Unauthorized
+ *
+ * components:
+ *    schemas:
+ *      profile:
+ *        type: object
+ *       
+ *        properties:
+ *          firstname:
+ *            type: object
+ *            properties:
+ *               value: 
+ *                  type: string
+ *               save: 
+ *                  type: boolean
+ *          lastname:
+ *            type: object
+ *            properties:
+ *                value: 
+ *                   type: string
+ *                save:
+ *                    type: boolean
+ *          gender:
+ *            type: object
+ *            properties:
+ *                value: 
+ *                   type: string
+ *                   enum: ["Male", "Female"]
+ *                save:
+ *                    type: boolean
+ *          origin:
+ *            type: object
+ *            properties:
+ *                value: 
+ *                   type: string
+ *                save:
+ *                    type: boolean
+ *          profession:
+ *            type: object
+ *            properties:
+ *                value: 
+ *                   type: string
+ *                save:
+ *                    type: boolean
+ *          age:
+ *            type: object
+ *            properties:
+ *                value: 
+ *                   type: integer
+ *                save:
+ *                    type: boolean
+ *          identification_type:
+ *            type: object
+ *            properties:
+ *                value: 
+ *                   type: string
+ *                save:
+ *                    type: boolean
+ *          identification_number:
+ *            type: object
+ *            properties:
+ *                value: 
+ *                   type: string
+ *                save:
+ *                    type: boolean
+ */
+
+router.put('/:userId', udpateProfile);
+
```

Comment:
Here is an issue, you need to protect this route so that you will make sure that whoever access this route is logged in so that you are sure of having token in the header, otherwise when you hit this route without signing in the server hang and no any message displayed just to be aware of what's going on.....

Quick fix: 1. You can either  protect this route as i mention or
                 2. In your controller before decoding the token check it existence if  no token, provide a message just to avaid server hanging

Reply:
Thanks! I worked on it now it ok

Reply:
Have you tested again to confirm fix @SimonImanigirimpuhwe ?